### PR TITLE
Task server backend

### PR DESCRIPTION
**Pull request by athithyaaselvam**
## What changes were proposed in this pull request?

- Task server backend changes

## How was this patch tested?

##- (Please explain how this patch was tested. Ex: unit tests, manual tests)
##- (If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

### CODE REVIEW

File: apps/filebrowser/src/filebrowser/tasks.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    kwargs["progress"] = "100%"
+    kwargs["task_end"] = now.strftime("%Y-%m-%dT%H:%M:%S")
+    document_cleanup_task.update_state(task_id=task_id, state='SUCCESS', meta=kwargs)
+
+    return
+
+import os
+import shutil
+import psutil
+from datetime import datetime, timedelta
```
Comment (amitsrivastava) [inline apps/filebrowser/src/filebrowser/tasks.py:134]: Can possibly move these imports to the top.

File: apps/filebrowser/src/filebrowser/views.py [hunk 2]
```diff
@@ -1458,10 +1458,11 @@ def _create_response(request, _fs, result="success", data="Success"):
       'success': True,
       'uuid': _fs.qquuid,
       'status': 0,
-      'data': data
+      'data': data,
+      'task_id': _fs.qquuid
   }
 
-def perform_upload(request, *args, **kwargs):
+def perform_upload_task(request, *args, **kwargs):
```
Comment (amitsrivastava) [inline apps/filebrowser/src/filebrowser/views.py:1465]: Why rename the function? It seems that it still supports the non task server method.

File: desktop/core/src/desktop/api2.py [hunk 3]
```diff
@@ -688,6 +692,155 @@ def share_document(request):
...
+  # Extract the task name and params from the request
+  try:
+    data = json.loads(request.body)
+    task_name = data.get('taskName')
+    task_params = data.get('taskParams')
+  except json.JSONDecodeError as e:
+    return JsonResponse({'error': str(e)}, status=500)
+
+  if task_name == 'document cleanup':
+    keep_days = task_params.get('keep-days')
```
Comment (amitsrivastava) [inline desktop/core/src/desktop/api2.py:710]: add a default value too.

File: desktop/core/src/desktop/api2.py [hunk 4]
```diff
@@ -688,6 +692,155 @@ def share_document(request):
...
+def handle_submit(request):
+  # Extract the task name and params from the request
+  try:
+    data = json.loads(request.body)
+    task_name = data.get('taskName')
+    task_params = data.get('taskParams')
+  except json.JSONDecodeError as e:
+    return JsonResponse({'error': str(e)}, status=500)
+
+  if task_name == 'document cleanup':
```
Comment (amitsrivastava) [inline desktop/core/src/desktop/api2.py:709]: I would recommend creating all the task names as constants in tasks.py and skip the spaces in values.
Something like
```
DOCUMENT_CLEANUP = 'DOCUMENT_CLEANUP'
```

File: desktop/core/src/desktop/api2.py [hunk 5]
```diff
@@ -688,6 +692,155 @@ def share_document(request):
...
+  is_finalized = task.get('status') == 'SUCCESS'
+  is_running = task.get('status') == 'RUNNING'
+  is_failure = task.get('status') == 'FAILURE'
+  is_revoked = task.get('status') == 'REVOKED'
+
+  return JsonResponse({'isFinalized': is_finalized, 'isRunning': is_running, 'isFailure': is_failure, 'isRevoked': is_revoked})
+
+from django.http import JsonResponse
+from celery.app.control import Control
+from desktop.celery import app as celery_app
```
Comment (amitsrivastava) [inline desktop/core/src/desktop/api2.py:787]: Is there a reason to keep these imports here in between the file? Unless these is some dependency issue, the imports should be at the top. It helps with maintaining the imports.

File: desktop/core/src/desktop/celery.py [hunk 6]
```diff
@@ -25,13 +25,38 @@
...
 from desktop.conf import TASK_SERVER
-from desktop.settings import TIME_ZONE, INSTALLED_APPS
+from desktop.settings import TIME_ZONE, INSTALLED_APPS, CELERY_RESULT_BACKEND, CELERY_BROKER_URL
 
 
 # Set the default Django settings module for the 'celery' program.
 os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'desktop.settings')
 
-app = Celery('desktop')
+class MyCelery(Celery):
```
Comment (amitsrivastava) [inline desktop/core/src/desktop/celery.py:34]: Why `MyCelery`? Why not `HueCelery`?

**Comment by athithyaaselvam** [comment]
Moved the imports to the top and renamed MyCelery. We have split the PR into two parts and abandoning this PR.
1. [Adding Task server backend changes for Hue #3687](https://github.com/cloudera/hue/pull/3687)  
2. [Task server Docker changes and backend tasks changes #3688](https://github.com/cloudera/hue/pull/3688  )

**Pull request closed without merging**